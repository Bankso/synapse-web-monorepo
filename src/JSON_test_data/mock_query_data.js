let mockData = {
    columnModels: [{
        columnType: "ENTITYID",
        name: "id",
        id: "2510"
    }],
    facets: [
        {
            facetCount: 1,
            columnName: "name",
            facetValues: ['']
        }
    ],
    queryResult: {
        queryResults: {
            rows: [{
                rowId: "1",
                values: ["syn5604373"]
            }],
            headers: [{
                name: "header"
            }]
        }
    }
}

export default mockData