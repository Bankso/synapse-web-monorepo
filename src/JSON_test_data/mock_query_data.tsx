import { QueryResultBundle } from 'src/lib/utils/jsonResponses/Table/QueryResultBundle';
import { ColumnType } from 'src/lib/utils/jsonResponses/Table/ColumnType';

let mockData: QueryResultBundle = {
    concreteType: "a",
    columnModels: [{
        id: "2510",
        name: "id",
        defaultValue: "a",
        columnType: ColumnType.STRING,
        enumValues: ["a"],
        maximumSize: 5120,
        facetType: {enumeration: "a"}
    }],
    facets: [
        {
            concreteType: "a",
            facetType: {enumeration: "a"},
            columnName: "name",
            facetValues: [{value: "a", count: 1, isSelected: true}]
        }
    ],
    queryResult: {
        queryResults: {
            etag: "s",
            rows: [{
                rowId: 1,
                values: ["syn5604373"],
                versionNumber: 123,
                etag: "a"
            }],
            headers: [{
                name: "header",
                columnType: ColumnType.STRING,
                id: "a"
            }]
        },
        nextPageToken: {token: "123"}
    }
}

export default mockData