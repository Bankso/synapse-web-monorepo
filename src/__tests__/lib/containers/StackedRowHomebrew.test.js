import React from 'react';
import {shallow, configure} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import StackedRowHomebrew from '../../../lib/containers/StackedRowHomebrew'
import {mockData} from '../../../JSON_test_data'

configure({ adapter: new Adapter() });

describe('it renders a chart without crashing', () => {

    it('renders a chart', async () => {
        const tree = await shallow(
            <StackedRowHomebrew
                data={mockData}             
                getLastQueryRequest={() => {return {query: {selectedFacets: {}}}}}
            ></StackedRowHomebrew>
        )
        expect(tree).toBeDefined()
        expect(tree).toMatchSnapshot()
    })

})