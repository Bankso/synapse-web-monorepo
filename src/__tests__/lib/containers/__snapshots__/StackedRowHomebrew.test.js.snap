// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders a chart without crashing renders a chart 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StackedRowHomebrew
    data={
        Object {
            "columnModels": Array [
              Object {
                "columnType": "STRING",
                "defaultValue": "a",
                "enumValues": Array [
                  "a",
                ],
                "facetType": "enumeration",
                "id": "2510",
                "maximumSize": 89,
                "name": "id",
              },
            ],
            "concreteType": "org.sagebionetworks.repo.model.table.QueryResultBundle",
            "facets": Array [
              Object {
                "columnName": "name",
                "concreteType": "a",
                "facetType": "enumeration",
                "facetValues": Array [
                  Object {
                    "count": 1,
                    "isSelected": true,
                    "value": "a",
                  },
                ],
              },
            ],
            "maxRowsPerPage": 100,
            "queryCount": 100,
            "queryResult": Object {
              "nextPageToken": Object {
                "token": "123",
              },
              "queryResults": Object {
                "etag": "s",
                "headers": Array [
                  Object {
                    "columnType": "STRING",
                    "id": "a",
                    "name": "header",
                  },
                ],
                "rows": Array [
                  Object {
                    "etag": "a",
                    "rowId": 1,
                    "values": Array [
                      "syn5604373",
                    ],
                    "versionNumber": 123,
                  },
                ],
              },
            },
            "selectColumns": Array [
              Object {
                "columnType": "STRING",
                "id": "a",
                "name": "name",
              },
            ],
            "sumFileSizes": Object {
              "greaterThan": false,
              "sumFileSizesBytes": 1000,
            },
          }
    }
    getLastQueryRequest={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="row SRC-center-text"
>
          <button
                    className="btn btn-default btn-sm SRC-floatRight"
                    onClick={[Function]}
          >
                    <FontAwesomeIcon
                              border={false}
                              className=""
                              fixedWidth={false}
                              flip={null}
                              icon="angle-right"
                              inverse={false}
                              listItem={false}
                              mask={null}
                              pull={null}
                              pulse={false}
                              rotation={null}
                              size={null}
                              spin={false}
                              symbol={false}
                              title=""
                              transform={null}
                    />
          </button>
          <button
                    className="btn btn-default btn-sm SRC-floatRight"
                    onClick={[Function]}
          >
                    <FontAwesomeIcon
                              border={false}
                              className=""
                              fixedWidth={false}
                              flip={null}
                              icon="angle-left"
                              inverse={false}
                              listItem={false}
                              mask={null}
                              pull={null}
                              pulse={false}
                              rotation={null}
                              size={null}
                              spin={false}
                              symbol={false}
                              title=""
                              transform={null}
                    />
          </button>
</div>,
        <div
          className="row SRC-bar-border SRC-bar-marginTop SRC-bar-border-top"
>
          <Measure
                    bounds={true}
                    onResize={[Function]}
          >
                    [Function]
          </Measure>
</div>,
        <div
          className="row SRC-bar-border SRC-bar-border-bottom"
>
          <p
                    className="SRC-noMargin SRC-padding-chart SRC-text-title"
          >
                    <strong>
                              <Unknown>
                                        <span
                                                  className="SRC-text-cap"
                                        >
                                                   
                                                   
                                        </span>
                                         : 
                                        <span>
                                                   
                                                   
                                        </span>
                              </Unknown>
                    </strong>
          </p>
          <p
                    className="SRC-noMargin SRC-padding-chart SRC-text-chart"
          >
                     
                    <a
                              href="https://www.synapse.org/#!Synapse:/tables/query/eyJzZWxlY3RlZEZhY2V0cyI6W3siY29uY3JldGVUeXBlIjoib3JnLnNhZ2ViaW9uZXR3b3Jrcy5yZXBvLm1vZGVsLnRhYmxlLkZhY2V0Q29sdW1uVmFsdWVzUmVxdWVzdCIsImZhY2V0VmFsdWVzIjpbbnVsbF19XX0="
                              target="_blank"
                    >
                               
                    </a>
                     
          </p>
</div>,
      ],
      "className": "container-fluid",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="btn btn-default btn-sm SRC-floatRight"
              onClick={[Function]}
>
              <FontAwesomeIcon
                            border={false}
                            className=""
                            fixedWidth={false}
                            flip={null}
                            icon="angle-right"
                            inverse={false}
                            listItem={false}
                            mask={null}
                            pull={null}
                            pulse={false}
                            rotation={null}
                            size={null}
                            spin={false}
                            symbol={false}
                            title=""
                            transform={null}
              />
</button>,
            <button
              className="btn btn-default btn-sm SRC-floatRight"
              onClick={[Function]}
>
              <FontAwesomeIcon
                            border={false}
                            className=""
                            fixedWidth={false}
                            flip={null}
                            icon="angle-left"
                            inverse={false}
                            listItem={false}
                            mask={null}
                            pull={null}
                            pulse={false}
                            rotation={null}
                            size={null}
                            spin={false}
                            symbol={false}
                            title=""
                            transform={null}
              />
</button>,
          ],
          "className": "row SRC-center-text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon="angle-right"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
/>,
              "className": "btn btn-default btn-sm SRC-floatRight",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "border": false,
                "className": "",
                "fixedWidth": false,
                "flip": null,
                "icon": "angle-right",
                "inverse": false,
                "listItem": false,
                "mask": null,
                "pull": null,
                "pulse": false,
                "rotation": null,
                "size": null,
                "spin": false,
                "symbol": false,
                "title": "",
                "transform": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon="angle-left"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
/>,
              "className": "btn btn-default btn-sm SRC-floatRight",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "border": false,
                "className": "",
                "fixedWidth": false,
                "flip": null,
                "icon": "angle-left",
                "inverse": false,
                "listItem": false,
                "mask": null,
                "pull": null,
                "pulse": false,
                "rotation": null,
                "size": null,
                "spin": false,
                "symbol": false,
                "title": "",
                "transform": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Measure
            bounds={true}
            onResize={[Function]}
>
            [Function]
</Measure>,
          "className": "row SRC-bar-border SRC-bar-marginTop SRC-bar-border-top",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bounds": true,
            "children": [Function],
            "onResize": [Function],
          },
          "ref": null,
          "rendered": [Function],
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="SRC-noMargin SRC-padding-chart SRC-text-title"
>
              <strong>
                            <Unknown>
                                          <span
                                                        className="SRC-text-cap"
                                          >
                                                         
                                                         
                                          </span>
                                           : 
                                          <span>
                                                         
                                                         
                                          </span>
                            </Unknown>
              </strong>
</p>,
            <p
              className="SRC-noMargin SRC-padding-chart SRC-text-chart"
>
               
              <a
                            href="https://www.synapse.org/#!Synapse:/tables/query/eyJzZWxlY3RlZEZhY2V0cyI6W3siY29uY3JldGVUeXBlIjoib3JnLnNhZ2ViaW9uZXR3b3Jrcy5yZXBvLm1vZGVsLnRhYmxlLkZhY2V0Q29sdW1uVmFsdWVzUmVxdWVzdCIsImZhY2V0VmFsdWVzIjpbbnVsbF19XX0="
                            target="_blank"
              >
                             
              </a>
               
</p>,
          ],
          "className": "row SRC-bar-border SRC-bar-border-bottom",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <strong>
                <Unknown>
                                <span
                                                className="SRC-text-cap"
                                >
                                                 
                                                 
                                </span>
                                 : 
                                <span>
                                                 
                                                 
                                </span>
                </Unknown>
</strong>,
              "className": "SRC-noMargin SRC-padding-chart SRC-text-title",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Unknown>
                  <span
                                    className="SRC-text-cap"
                  >
                                     
                                     
                  </span>
                   : 
                  <span>
                                     
                                     
                  </span>
</Unknown>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <span
                      className="SRC-text-cap"
>
                       
                       
</span>,
                    " : ",
                    <span>
                       
                       
</span>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        undefined,
                        " ",
                      ],
                      "className": "SRC-text-cap",
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      undefined,
                      " ",
                    ],
                    "type": "span",
                  },
                  " : ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        undefined,
                        " ",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      undefined,
                      " ",
                    ],
                    "type": "span",
                  },
                ],
                "type": Symbol(react.fragment),
              },
              "type": "strong",
            },
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                " ",
                <a
                  href="https://www.synapse.org/#!Synapse:/tables/query/eyJzZWxlY3RlZEZhY2V0cyI6W3siY29uY3JldGVUeXBlIjoib3JnLnNhZ2ViaW9uZXR3b3Jrcy5yZXBvLm1vZGVsLnRhYmxlLkZhY2V0Q29sdW1uVmFsdWVzUmVxdWVzdCIsImZhY2V0VmFsdWVzIjpbbnVsbF19XX0="
                  target="_blank"
>
                   
</a>,
                " ",
              ],
              "className": "SRC-noMargin SRC-padding-chart SRC-text-chart",
            },
            "ref": null,
            "rendered": Array [
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    undefined,
                    " ",
                    undefined,
                  ],
                  "href": "https://www.synapse.org/#!Synapse:/tables/query/eyJzZWxlY3RlZEZhY2V0cyI6W3siY29uY3JldGVUeXBlIjoib3JnLnNhZ2ViaW9uZXR3b3Jrcy5yZXBvLm1vZGVsLnRhYmxlLkZhY2V0Q29sdW1uVmFsdWVzUmVxdWVzdCIsImZhY2V0VmFsdWVzIjpbbnVsbF19XX0=",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                  " ",
                  undefined,
                ],
                "type": "a",
              },
              " ",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="row SRC-center-text"
>
            <button
                        className="btn btn-default btn-sm SRC-floatRight"
                        onClick={[Function]}
            >
                        <FontAwesomeIcon
                                    border={false}
                                    className=""
                                    fixedWidth={false}
                                    flip={null}
                                    icon="angle-right"
                                    inverse={false}
                                    listItem={false}
                                    mask={null}
                                    pull={null}
                                    pulse={false}
                                    rotation={null}
                                    size={null}
                                    spin={false}
                                    symbol={false}
                                    title=""
                                    transform={null}
                        />
            </button>
            <button
                        className="btn btn-default btn-sm SRC-floatRight"
                        onClick={[Function]}
            >
                        <FontAwesomeIcon
                                    border={false}
                                    className=""
                                    fixedWidth={false}
                                    flip={null}
                                    icon="angle-left"
                                    inverse={false}
                                    listItem={false}
                                    mask={null}
                                    pull={null}
                                    pulse={false}
                                    rotation={null}
                                    size={null}
                                    spin={false}
                                    symbol={false}
                                    title=""
                                    transform={null}
                        />
            </button>
</div>,
          <div
            className="row SRC-bar-border SRC-bar-marginTop SRC-bar-border-top"
>
            <Measure
                        bounds={true}
                        onResize={[Function]}
            >
                        [Function]
            </Measure>
</div>,
          <div
            className="row SRC-bar-border SRC-bar-border-bottom"
>
            <p
                        className="SRC-noMargin SRC-padding-chart SRC-text-title"
            >
                        <strong>
                                    <Unknown>
                                                <span
                                                            className="SRC-text-cap"
                                                >
                                                             
                                                             
                                                </span>
                                                 : 
                                                <span>
                                                             
                                                             
                                                </span>
                                    </Unknown>
                        </strong>
            </p>
            <p
                        className="SRC-noMargin SRC-padding-chart SRC-text-chart"
            >
                         
                        <a
                                    href="https://www.synapse.org/#!Synapse:/tables/query/eyJzZWxlY3RlZEZhY2V0cyI6W3siY29uY3JldGVUeXBlIjoib3JnLnNhZ2ViaW9uZXR3b3Jrcy5yZXBvLm1vZGVsLnRhYmxlLkZhY2V0Q29sdW1uVmFsdWVzUmVxdWVzdCIsImZhY2V0VmFsdWVzIjpbbnVsbF19XX0="
                                    target="_blank"
                        >
                                     
                        </a>
                         
            </p>
</div>,
        ],
        "className": "container-fluid",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="btn btn-default btn-sm SRC-floatRight"
                onClick={[Function]}
>
                <FontAwesomeIcon
                                border={false}
                                className=""
                                fixedWidth={false}
                                flip={null}
                                icon="angle-right"
                                inverse={false}
                                listItem={false}
                                mask={null}
                                pull={null}
                                pulse={false}
                                rotation={null}
                                size={null}
                                spin={false}
                                symbol={false}
                                title=""
                                transform={null}
                />
</button>,
              <button
                className="btn btn-default btn-sm SRC-floatRight"
                onClick={[Function]}
>
                <FontAwesomeIcon
                                border={false}
                                className=""
                                fixedWidth={false}
                                flip={null}
                                icon="angle-left"
                                inverse={false}
                                listItem={false}
                                mask={null}
                                pull={null}
                                pulse={false}
                                rotation={null}
                                size={null}
                                spin={false}
                                symbol={false}
                                title=""
                                transform={null}
                />
</button>,
            ],
            "className": "row SRC-center-text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon="angle-right"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
/>,
                "className": "btn btn-default btn-sm SRC-floatRight",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "",
                  "fixedWidth": false,
                  "flip": null,
                  "icon": "angle-right",
                  "inverse": false,
                  "listItem": false,
                  "mask": null,
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": null,
                  "spin": false,
                  "symbol": false,
                  "title": "",
                  "transform": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon="angle-left"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
/>,
                "className": "btn btn-default btn-sm SRC-floatRight",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "",
                  "fixedWidth": false,
                  "flip": null,
                  "icon": "angle-left",
                  "inverse": false,
                  "listItem": false,
                  "mask": null,
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": null,
                  "spin": false,
                  "symbol": false,
                  "title": "",
                  "transform": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Measure
              bounds={true}
              onResize={[Function]}
>
              [Function]
</Measure>,
            "className": "row SRC-bar-border SRC-bar-marginTop SRC-bar-border-top",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bounds": true,
              "children": [Function],
              "onResize": [Function],
            },
            "ref": null,
            "rendered": [Function],
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="SRC-noMargin SRC-padding-chart SRC-text-title"
>
                <strong>
                                <Unknown>
                                                <span
                                                                className="SRC-text-cap"
                                                >
                                                                 
                                                                 
                                                </span>
                                                 : 
                                                <span>
                                                                 
                                                                 
                                                </span>
                                </Unknown>
                </strong>
</p>,
              <p
                className="SRC-noMargin SRC-padding-chart SRC-text-chart"
>
                 
                <a
                                href="https://www.synapse.org/#!Synapse:/tables/query/eyJzZWxlY3RlZEZhY2V0cyI6W3siY29uY3JldGVUeXBlIjoib3JnLnNhZ2ViaW9uZXR3b3Jrcy5yZXBvLm1vZGVsLnRhYmxlLkZhY2V0Q29sdW1uVmFsdWVzUmVxdWVzdCIsImZhY2V0VmFsdWVzIjpbbnVsbF19XX0="
                                target="_blank"
                >
                                 
                </a>
                 
</p>,
            ],
            "className": "row SRC-bar-border SRC-bar-border-bottom",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <strong>
                  <Unknown>
                                    <span
                                                      className="SRC-text-cap"
                                    >
                                                       
                                                       
                                    </span>
                                     : 
                                    <span>
                                                       
                                                       
                                    </span>
                  </Unknown>
</strong>,
                "className": "SRC-noMargin SRC-padding-chart SRC-text-title",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Unknown>
                    <span
                                        className="SRC-text-cap"
                    >
                                         
                                         
                    </span>
                     : 
                    <span>
                                         
                                         
                    </span>
</Unknown>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      <span
                        className="SRC-text-cap"
>
                         
                         
</span>,
                      " : ",
                      <span>
                         
                         
</span>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          " ",
                          undefined,
                          " ",
                        ],
                        "className": "SRC-text-cap",
                      },
                      "ref": null,
                      "rendered": Array [
                        " ",
                        undefined,
                        " ",
                      ],
                      "type": "span",
                    },
                    " : ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          " ",
                          undefined,
                          " ",
                        ],
                      },
                      "ref": null,
                      "rendered": Array [
                        " ",
                        undefined,
                        " ",
                      ],
                      "type": "span",
                    },
                  ],
                  "type": Symbol(react.fragment),
                },
                "type": "strong",
              },
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  <a
                    href="https://www.synapse.org/#!Synapse:/tables/query/eyJzZWxlY3RlZEZhY2V0cyI6W3siY29uY3JldGVUeXBlIjoib3JnLnNhZ2ViaW9uZXR3b3Jrcy5yZXBvLm1vZGVsLnRhYmxlLkZhY2V0Q29sdW1uVmFsdWVzUmVxdWVzdCIsImZhY2V0VmFsdWVzIjpbbnVsbF19XX0="
                    target="_blank"
>
                     
</a>,
                  " ",
                ],
                "className": "SRC-noMargin SRC-padding-chart SRC-text-chart",
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      undefined,
                      " ",
                      undefined,
                    ],
                    "href": "https://www.synapse.org/#!Synapse:/tables/query/eyJzZWxlY3RlZEZhY2V0cyI6W3siY29uY3JldGVUeXBlIjoib3JnLnNhZ2ViaW9uZXR3b3Jrcy5yZXBvLm1vZGVsLnRhYmxlLkZhY2V0Q29sdW1uVmFsdWVzUmVxdWVzdCIsImZhY2V0VmFsdWVzIjpbbnVsbF19XX0=",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                    " ",
                    undefined,
                  ],
                  "type": "a",
                },
                " ",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
